guides/introduction/Embedded Schemas.md

```diff
@@ -1,89 +1,86 @@
 # Embedded Schemas
 
-Embedded schemas are a feature in Ecto that allows you to define and validate structured data. This data can reside in-memory through structs or, with a migration, be stored in a database field. Embedded schemas work with maps and arrays.
+Embedded schemas allow you to define and validate structured data that is nested within another struct. This data can live exclusively in memory, or can be stored in the database.
 
-They're great for:
+Some use cases for embedded schemas include:
 
-- Storing additional data about an entity without modifying your schema.
-- Storing additional data about an entity without tracking foreign key/many-to-many relationships
-- Validating arbitrary data structures
+- Embedding flexible data that changes often, like a map of user preferences inside a User schema.
 
-Example use cases:
+- Embedding simple data that you want to track and validate, but where maintaining a separate table and tracking foreign keys and many-to-many relationships is unnecessary, like a list of product images.
 
-- User profiles: storing additional information like profile pictures, settings
-- Shop products: storing additional product images
+- When you want nuanced control over the tracking and validation of a complex struct, like if you wanted a changeset for address validation, but didn't want a separate address table.
 
-While embedded schemas require an initial migration to create the field, any subsequent modifications to the data structure don't.
+- When using document storage databases, and you want to interact with and manipulate embedded documents.
 
-This feature is great for situations where you'd like to store data associated to a specific entity, but a relation or new table would be overkill. You can even write changesets to perform validations on this data! The downside, however, is as you change the embedded schema overtime, early data won't have fields that you added later on.
+These above use cases have a few themes in common. They all have nested data, and it makes sense for that nested data to be (1) bundled together into a sub-entity, and (2) changeset validations on that sub-entity are desireable.
```

Comment:
As mentioned above, it is not always about nesting, so we should revisit this. :)

Reply:
I'm happy to abandon the "nesting" phraseology. Thank you for the feedback <3

I think the thing I want to succinctly articulate is the following: The schema is representing something. That something is not well represented by a flat list of data fields; it is better represented by structured data wherein data lives at multiple layers. From this fact, it logically follows that you will want to handle those sub-structs using their own schemas and changesets. And here comes embedded schemas to save the day!

Do you agree with that core narrative? If so, then I will continue trying to find a simple and succinct way to articulate it. Otherwise, I need to keep reading the docs XD

Reply:
That's a good chunk of the narrative but the truth is that embedded schemas can completely exist on their own, even if not nested anywhere, as they are useful to validate data that is not persisted anywhere. This guide has a brief example of it: https://github.com/elixir-ecto/ecto/blob/master/guides/howtos/Data%20mapping%20and%20validation.md

Reply:
That is _really_ helpful feedback. Thank you!! You give a similar example here, which was a nice explanation as well: https://hexdocs.pm/ecto/Ecto.Schema.html#module-example . 

I reckon others get confused by the name `embedded_schema` the same way I did, and focus solely on the "nesting" aspects of it at first. I think I'd like to call that out somewhere in this intro to help future readers, if you think that's a good idea.

In trying to understand why you couldn't use a regular schema in places where you use embedded_schema, I realized that the programmatic difference between them is really that schemas have the __meta__ map that keeps track of persistence state. So it's really that `embedded_schema` == `persistence_agnostic_schema`, right? You get all the upside of schemas without worry about persistence. And if you _do_ want to worry about persistence, you do so via a parent schema, or manually like you did in the Data mapping and validation guide.

Reply:
You got it!

Reply:
<3 I'm adding the section at the ended of the "extracting" section to mention this, so resolving it here.